core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -905,22 +905,22 @@ class KafkaApis(val requestChannel: RequestChannel,
         if (header.apiVersion == 0) {
           val (authorizedPartitions, unauthorizedPartitions) = offsetFetchRequest.partitions.asScala
             .partition(authorizeTopicDescribe)
+          // authorizedAndExistingPartitions: authorized partitions that actually exist in the cluster, and need their offset retrieved
+          // nonExistingPartitions: partitions that do not exist in the cluster, and should get an unknown partition error
+          val (authorizedAndExistingPartitions, nonExistingPartitions) =
```

Comment:
Thanks. It makes sense. The redundancy bothers me too. I'll try to address it in the next update.